tests/runner/test_utils.py

```diff
@@ -60,60 +69,95 @@ def deployment_openapi_schema(schema_item):
 
 
 @pytest.fixture
-def openapi_schema():
-    return {"components": {"schemas": {}}}
-
-
-def test_inject_schemas_into_openapi(mock_app, deployment_schemas):
-    augmented_schema = inject_schemas_into_openapi(mock_app, deployment_schemas)
-    assert "Model1" in augmented_schema["components"]["schemas"]
-    assert "Model2" in augmented_schema["components"]["schemas"]
-    assert augmented_schema["components"]["schemas"]["Model1"]["type"] == "object"
-    assert augmented_schema["components"]["schemas"]["Model2"]["type"] == "object"
-    assert (
-        augmented_schema["components"]["schemas"]["Model1"]["properties"]["field1"][
-            "type"
-        ]
-        == "string"
-    )
-    assert (
-        augmented_schema["components"]["schemas"]["Model2"]["properties"]["field2"][
-            "type"
-        ]
-        == "integer"
-    )
-
-
-def test_merge_definitions(deployment_schemas, openapi_schema):
-    merge_definitions(deployment_schemas, openapi_schema)
-    assert "Model1" in openapi_schema["components"]["schemas"]
-    assert "Model2" in openapi_schema["components"]["schemas"]
-    assert openapi_schema["components"]["schemas"]["Model1"]["type"] == "object"
-    assert openapi_schema["components"]["schemas"]["Model2"]["type"] == "object"
-    assert (
-        openapi_schema["components"]["schemas"]["Model1"]["properties"]["field1"][
-            "type"
-        ]
-        == "string"
-    )
-    assert (
-        openapi_schema["components"]["schemas"]["Model2"]["properties"]["field2"][
-            "type"
-        ]
-        == "integer"
-    )
-
-
-def test_update_refs_in_schema(schema_item):
-    update_refs_in_schema(schema_item, "#/components/schemas/")
-    assert schema_item["$ref"] == "#/components/schemas/Model1"
-
-
-def test_update_refs_to_components(deployment_openapi_schema):
-    update_refs_to_components(deployment_openapi_schema)
-    assert (
-        deployment_openapi_schema["paths"]["/path"]["get"]["requestBody"]["content"][
-            "application/json"
-        ]["schema"]["$ref"]
-        == "#/components/schemas/Model1"
-    )
+def nested_schema_with_refs():
+    return {
+        "definitions": {
+            "NestedModel": {
+                "type": "object",
+                "properties": {"nestedField": {"$ref": "#/definitions/Model1"}},
+            }
+        },
+        "paths": {
+            "/nested": {
+                "get": {
+                    "requestBody": {
+                        "content": {
+                            "application/json": {
+                                "schema": {"$ref": "#/definitions/NestedModel"}
+                            }
+                        }
+                    }
+                }
+            }
+        },
+    }
+
+
+@pytest.fixture
+def augmented_openapi_schema(deployment_schemas, openapi_schema):
+    merged_schema = merge_definitions(deployment_schemas, openapi_schema)
+    return update_refs_to_components(merged_schema)
```

Comment:
Using this fixture as the expected value for the test below doesn't tell me much and tests the implementation more than the interface. Can you update this fixture to return a dictionary that matches the expected output?

Reply:
switched to a more literal fixture in [752a98f](https://github.com/PrefectHQ/prefect/pull/11333/commits/752a98f1f103b4616299fe914c1bc420f734a7a0)